### DISCUSSION TITLE
Is rounded background modules coming any time?

### DISCUSSION THREAD

**Discussion by acidburnmonkey** [polybar/polybar#2916] [General]
Also if somebody can point me to the files that handle background and module foreground so i can look at it.

**Comment by patrick96** [comment]
Probably not, the current system just can't support that. The only way you can get rounded modules is by adding characters that look rounded, for example with powerline symbols:

[![](https://raw.githubusercontent.com/ryanoasis/powerline-extra-symbols/master/img/preview.png)](https://github.com/ryanoasis/powerline-extra-symbols)

The code is a bit all over the place. The main rendering is done in [`renderer.cpp`](https://github.com/polybar/polybar/blob/master/src/components/renderer.cpp) which gets instruction on what to render from [`dispatch.cpp`](https://github.com/polybar/polybar/blob/master/src/tags/dispatch.cpp). At that point, all information about where modules are is already lost, there is just one big string of [format tags](https://github.com/polybar/polybar/wiki/Formatting#format-tags) that was stitched together from all modules.
